src/keystores/base_keystore.rs

```diff
@@ -0,0 +1,237 @@
+use std::{collections::HashMap, fs, os::unix::fs::PermissionsExt};
+
+use aes::{cipher::{generic_array::GenericArray, KeyIvInit, StreamCipher}, Aes128};
+use ctr::Ctr128BE;
+use num_traits::ToPrimitive;
+use rand::Rng;
+use serde::{Deserialize, Serialize};
+use serde_json::{Map, Value};
+use sha2::{Digest, Sha256};
+use unicode_normalization::UnicodeNormalization;
+use uuid::Uuid;
+use std::io::Write;
```

Comment:
nit can we separate the std imports, the libraries and the crate imports?

Reply:
here and elsewhere

Reply:
pushed missing code.